app/tests/test_collections_endpoint.py

```diff
@@ -84,3 +86,130 @@ def test_collections_limit_query(self):
+        )
+        response_json = response.json()
+        self.assertEqual(response.status_code, 400, msg=get_http_error_description(response_json))
+
+    def test_collections_put(self):
+        payload_json = self.sample_collections['valid_collection_set_2']
+
+        # the dataset to update does not exist yet
+        response = self.client.put(
+            f"/{API_BASE}/collections/{payload_json['id']}",
+            data=payload_json,
+            content_type='application/json'
+        )
+        response_json = response.json()
+        self.assertEqual(response.status_code, 404, msg=get_http_error_description(response_json))
+
+        # POST data
+        self.client.post(
+            f"/{API_BASE}/collections", data=payload_json, content_type='application/json'
+        )
+
+        payload_json['title'] = "The Swallows"
+        response = self.client.put(
+            f"/{API_BASE}/collections/{payload_json['id']}",
+            data=payload_json,
+            content_type='application/json'
+        )
+        response_json = response.json()
+
+        self.assertEqual(response.status_code, 200, msg=get_http_error_description(response_json))
+        self.assertEqual(response_json['title'], payload_json['title'])
+
+        # is it persistent?
+        response = self.client.get(
+            f"/{API_BASE}/collections/{payload_json['id']}",
+            data=payload_json,
+            content_type='application/json'
+        )
+
+        response_json = response.json()
+
+        self.assertEqual(response.status_code, 200, msg=get_http_error_description(response_json))
+        self.assertEqual(response_json['title'], payload_json['title'])
+
+    def test_collection_put_change_id(self):
+        payload_json = self.sample_collections['valid_collection_set_3']
+        logger.debug(self.collections[0].name)
+        response = self.client.put(
+            f"/{API_BASE}/collections/{self.collections[0].name}",
+            data=payload_json,
+            content_type='application/json'
+        )
+        response_json = response.json()
+        self.assertEqual(response.status_code, 200, msg=get_http_error_description(response_json))
+
+        # check if id changed
+        response = self.client.get(f"/{API_BASE}/collections/{payload_json['id']}")
+        response_json = response.json()
+        self.assertEqual(response_json['id'], payload_json['id'])
+
+        # the old collection shouldn't exist any more
+        response = self.client.get(f"/{API_BASE}/collections/{self.collections[0].name}")
+        response_json = response.json()
+        self.assertEqual(response.status_code, 404, msg=get_http_error_description(response_json))
+
+    def test_collection_put_none_values(self):
+        collection_name = self.collections[1].name
+        payload_json = self.sample_collections['valid_min_collection_set_2']
+        payload_json['id'] = collection_name
```

Comment:
Maybe test name should be `test_collection_put_remove_title_provider` or `test_collection_put_remove_optional_fields`. Also before the PUT some assert to make sure that `title` and `provider` are not already empty would increase the unittest safety.